src/js/controllers/mapController.ts

```diff
@@ -1142,6 +1145,295 @@ export class MapController {
+      default:
+        break;
+    }
+  }
+
+  setVIIRSDefinedRange(layer: any, sublayerType: string): void {
+    if (!this._map) {
+      return;
+    }
+    const VIIRS24 = layer.sublayers.items.filter((sublayer: Sublayer) =>
+      sublayer.title.includes('24 hrs')
+    );
+    const VIIRS48 = this._map?.findLayerById('VIIRS48');
+    const VIIRS72 = this._map?.findLayerById('VIIRS72');
+    const VIIRS7D = this._map?.findLayerById('VIIRS7D');
+
+    switch (sublayerType) {
+      case '24 hrs':
+        {
+          VIIRS48.visible = false;
+          VIIRS72.visible = false;
+          VIIRS7D.visible = false;
+          VIIRS24.visible = true;
+        }
+        break;
+      case '48 hrs':
+        {
+          VIIRS24.visible = false;
+          VIIRS72.visible = false;
+          VIIRS7D.visible = false;
+          VIIRS48.visible = true;
+        }
+        break;
+      case '72 hrs':
+        {
+          VIIRS24.visible = false;
+          VIIRS48.visible = false;
+          VIIRS7D.visible = false;
+          VIIRS72.visible = true;
+        }
+        break;
+      case '7 days':
+        {
+          VIIRS24.visible = false;
+          VIIRS48.visible = false;
+          VIIRS72.visible = false;
+          VIIRS7D.visible = true;
+        }
+        break;
+      default:
+        break;
+    }
+  }
+
+  resetDefinedDateRange(layerID: string): void {
+    if (!this._map) {
+      return;
+    }
+
+    const layer = (this._map.allLayers as any).items.filter(
+      (layer: LayerProps) => layer.id === layerID
+    )[0];
+
+    if (!layer.sublayers) {
+      return;
+    }
+
+    const sublayer24 = layer.sublayers.items.filter((sublayer: Sublayer) =>
+      sublayer.title.includes('24 hrs')
+    );
+
+    sublayer24.visible = false;
+
+    if (layer.id.includes('VIIRS')) {
+      const VIIRS48 = this._map?.findLayerById('VIIRS48');
+      const VIIRS72 = this._map?.findLayerById('VIIRS72');
+      const VIIRS7D = this._map?.findLayerById('VIIRS7D');
+
+      VIIRS48.visible = false;
+      VIIRS72.visible = false;
+      VIIRS7D.visible = false;
+    } else if (layer.id.includes('MODIS')) {
+      const MODIS48 = this._map?.findLayerById('MODIS48');
+      const MODIS72 = this._map?.findLayerById('MODIS72');
+      const MODIS7D = this._map?.findLayerById('MODIS7D');
+
+      MODIS48.visible = false;
+      MODIS72.visible = false;
+      MODIS7D.visible = false;
+    }
+  }
+
+  updateMODISorVIIRSOpacity(layerID: string, opacity: number): void {
+    if (!this._map) {
+      return;
+    }
+
+    const layer = (this._map.allLayers as any).items.filter(
+      (layer: LayerProps) => layer.id === layerID
+    )[0];
+
+    if (!layer.sublayers) {
+      return;
+    }
+
+    const sublayer24 = layer.sublayers.items.filter((sublayer: Sublayer) =>
+      sublayer.title.includes('24 hrs')
+    );
+
+    sublayer24.opacity = opacity;
+
+    if (layerID.includes('VIIRS')) {
+      const VIIRS48 = this._map?.findLayerById('VIIRS48');
```

Comment:
let's create an config object or array in layer configs helper that has listed all VIIRS fires layer IDS. That way we can use that same array, to iterate over when we create these layers or look for them by. Let me know if that does not make sense, happy to pair program.